doc/copp/Preserve_COPP_tables_to_improve_fast-reboot.md

```diff
@@ -0,0 +1,46 @@
+# Introduction
+
+The scope of this document is to provide the requirements and a high-level design proposal for preserving the contents of the CoPP (Sonic Control Plane Policing) tables during reboot for faster LAG creation in order to improve fast-reboot's dataplane downtime.
+
+# Requirements
+
+The following are the high level requirements for preserving CoPP tables contents during reboot.
+
+1. CoPP Tables shouldn't be cleared from APP DB by DB migrator.
+2. coppmgr mergeConfig logic should be enhanced to:
+    1. Ignore setting existing entries.
+    2. Overwrite entry when value differs, use value from default init file merged with user configuration.
+    3. Delete entries with keys that are not supported in the copp default initialization (backward compatibility).
+    4. In case an entry exists in user's configuration file or in the default json initialization file while missing from the preserved CoPP table it will be add to the table as a new entry.
+
+# Design Proposal
+
+## Current behavior
+
+In the current implementation DB migrator clears CoPP tables contents and it is being initialized with default values at startup. This process takes some time and leads to that LACP are being missed shortly after reboot since LACP trap is not set yet, thus delaying LAG creation and extending dataplane downtime during fast-reboot.
+
+## Proposed behavior
+
+With the new proposal, the CoPP tables contents will be preserved during reboot, i.e. they won't be cleared by DB migrator. Then, initializing CoPP tables in startup phase for any key-value entry it will be checked if such entry exists, in case it does, the entry will be ignored. In case there is an entry with the same key but with different value preserved from prior reboot the existing entry will be deleted and a new entry will be added to the CoPP tables with the key and the new value.
```

Comment:
can you please clarify if current startup coppmgr logic would still be performed or only new merge logic in case when app-db is not cleared in migrator

Reply:
This is just an enhancement of the current startup coppmgr logic. It uses the existing merge logic and adds a new check for duplicates or overwrites.

Reply:
ok, looking at the code it is clear